test/jasmine/tests/transform_sort_test.js

```diff
@@ -241,4 +242,102 @@ describe('Test sort transform interactions:', function() {
         .catch(fail)
         .then(done);
     });
+
+    it('does not preserve hover/click `pointNumber` value', function(done) {
```

Comment:
@n-riesco @alexcjohnson IMPORTANT: event data `pointNumber` does not correspond to array index in `gd.data` after a sort transform. The same is true for filter transform with `preservegaps: false`. This is an unfortunate consequence of the current transform pipeline. Using `ids` become crucial here. That is, as presented in this test case:

```js
var pt = eventData.points[0]

pt.fullData.ids[pt.pointNumber] 
// => is preserved after a sort transform
```

We could eventually add a `originalPointNumber` field to the event data to solve this problem. Moreover, we should really add an `id` field to the hover & click event data and make the `ids` data array more ubiquitous (it is only available for scatter traces at the moment).